jre_emul/android/platform/libcore/ojluni/src/main/java/java/io/OutputStreamWriter.java

```diff
@@ -308,17 +307,15 @@ public void write(int oneChar) throws IOException {
      *             {@code str}.
      */
     @Override
+    //TODO: use StreamEncoder per the update
     public void write(String str, int offset, int count) throws IOException {
         synchronized (lock) {
-            if (count < 0) {
-                throw new StringIndexOutOfBoundsException(str, offset, count);
+            if ((offset | count) < 0 || offset > str.length() - count) {
+                throw new StringIndexOutOfBoundsException(str + "offset=" + offset + "formatElementIndex=" + count);
             }
             if (str == null) {
```

Comment:
This test is redundant since the str.length() call above would through the NullPointerException, but there's no need to fix it here since it will be replaced by the StreamEncoder call.

Reply:
Woops, missed that. Thank you